src/main/java/ch/njol/skript/expressions/ExprWorld.java

```diff
@@ -106,4 +110,18 @@ public boolean setTime(final int time) {
 		return super.setTime(time, getExpr(), PlayerTeleportEvent.class);
 	}
 	
+	@Override
+	public void change(Event e, Object[] delta, Changer.ChangeMode mode){
+		if (getExpr().getAll(e) instanceof Location[] && delta != null)
+			for (final Location loc : (Location[]) getExpr().getAll(e)) {
+				loc.setWorld((World) delta[0]);
+			}	
+	}
+	
+	@Override
```

Comment:
Another forgotten @Nullable annotation here. 2nd and last reason why it didn't compile.